elastalert/schema.yaml

```diff
@@ -0,0 +1,187 @@
+            wildcard: {type: object}
+            query_string:
+              additionalProperties: true
+              properties:
+                query: {type: string}
+              type: object
+          type: object
+      # 'null' type is for backward-compatibility with existing examples
+      type: ['object', 'null']
+
+required: [name, type, index, alert, es_port, es_host]
+type: object
+properties:
+
+  # Common Settings
+  es_host: {type: string}
+  es_port: {type: integer}
+  index: {type: string}
+  name: {type: string}
+  type: {type: string}
+  alert: {type: string}
+
+  use_ssl: {type: boolean}
+  es_username: {type: string}
+  es_password: {type: string}
+  use_strftime_index: {type: boolean}
+
+  # Optional Settings
+  aggregation: *timeframe
+  realert: *timeframe
+  exponential_realert: *timeframe
+
+  buffer_time: *timeframe
+  query_delay: *timeframe
+  max_query_size: {type: integer}
+
+  filter :
+    type: array
+    items: *filter
+
+  include: {type: array, items: {type: string}}
+  top_count_keys: {type: array, items: {type: string}}
+  top_count_number: {type: integer}
+  raw_count_keys: {type: boolean}
+  generate_kibana_link: {type: boolean}
+  kibana_dashboard: {type: string}
+  use_kibana_dashboard: {type: string}
+  use_local_time: {type: boolean}
+  match_enhancements: {type: array, items: {type: string}}
+
+  ### Blacklist Rule
+  compare_key: {type: string}
+  blacklist: {type: array, items: {type: string}}
+
+  ### Whitelist Rule
+  # compare_key
+  whitelist: {type: array, items: {type: string}}
+  ignore_null: {type: boolean}
+
+  ### Change Rule
+  # ignore_null required
+  query_key: {type: string}
+  timeframe: *timeframe
+
+  ### Frequency Rule
+  num_events: {type: integer}
+  # timeframe
+  use_count_query: {type: boolean}
+  doc_type: {type: string}
+  use_terms_query: {type: boolean}
+  terms_size: {type: integer}
+  query_key: {type: string}
+
+  ### Spike Rule
+  spike_height: {type: number}
+  spike_type: {enum: ["up", "down", "both"]}
+  # timeframe
+  # query_key
+  # use_count_query
+  # doc_type
+  # use_terms_query
+  # terms_size
+  alert_on_new_data: {type: boolean}
+  threshold_ref: {type: integer}
+  threshold_cur: {type: integer}
+
+  ### Flatline
+  threshold: {type: integer}
+  # Use_count_query
+  # doc_type
+
+  ### New Term
+  fields: {type: string}
+  terms_window_size: *timeframe
+  alert_on_missing_field: {type: boolean}
+  # use_terms_query
+
+  # Alerts
+  alert: *arrayOfString
+  alert_subject: {type: string} # Python format string
+  alert_subject_args: {type: array, items: {type: string}}
+
+  # Alert Content
+  alert_text: {type: string} # Python format string
+  alert_text_args: {type: array, items: {type: string}}
+  alert_text_type: {enum: [alert_text_only, exclude_fields]}
+  timestamp_field: {type: string}
+  field: {}
+
+  ### Commands
+  command: *arrayOfString
+  pipe_match_json: {type: boolean}
+
+  ### Email
+  email: *arrayOfString
```

Comment:
This can also be just a string. I think single recipient is common enough that it should take both.